drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioRunnerServiceImpl.java

```diff
@@ -46,50 +48,67 @@
 
     @Inject
     private Event<TestResultMessage> defaultTestResultMessageEvent;
+    @Inject
+    private ScenarioLoader scenarioLoader;
 
     private ScenarioRunnerProvider runnerSupplier = null;
 
     @Override
     public void runAllTests(final String identifier,
                             final Path path) {
-
-        defaultTestResultMessageEvent.fire(
-                new TestResultMessage(
-                        identifier,
-                        1,
-                        1,
-                        new ArrayList<>()));
+        runAllTests(identifier,
+                    path,
+                    defaultTestResultMessageEvent);
     }
 
     @Override
     public void runAllTests(final String identifier,
                             final Path path,
                             final Event<TestResultMessage> customTestResultEvent) {
+        final CollectorEvent collectorEvent = new CollectorEvent();
 
-        customTestResultEvent.fire(
-                new TestResultMessage(
-                        identifier,
-                        1,
-                        1,
-                        new ArrayList<>()));
+        for (Map.Entry<Path, ScenarioSimulationModel> entry : scenarioLoader.loadScenarios(path).entrySet()) {
+
+            final ScenarioSimulationModel scenarioSimulationModel = entry.getValue();
+
+            runTest(identifier,
```

Comment:
@Rikkola 
It seems that customTestResultEvent.fire(...); gets invoked once for each entry (inside the runTest method) and then another time at the end of the loop: is this actually wanted? How/where are managed all that different "events" ?

Reply:
Inside runTest method it fires the event to the CollectorEvent that does not publish the event outside of this class. Then here a new event is created based on the event and published.